src/main/java/com/limechain/client/FullNode.java

```diff
@@ -69,7 +71,14 @@ public void start() {
 
         // Start syncing
         log.log(Level.INFO, "Node successfully connected to a peer! Sync can start!");
-        AppBean.getBean(WarpSyncMachine.class).start();
+
+        CliArguments args = AppBean.getBean(CliArguments.class);
+        switch (args.syncMode()){
+            case FULL -> AppBean.getBean(FullSyncMachine.class).start();
+            case WARP -> AppBean.getBean(WarpSyncMachine.class).start();
+            default -> AppBean.getBean(WarpSyncMachine.class).start(); //This statement should be impossible to reach
+            //but it's here just in case
```

Comment:
If the default should be unreachable I think it would be best if you make it throw an exception and have its message contain the `args.syncMode()` info.